gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -145,12 +142,14 @@ public int activeRequests() {
   StorageStubProvider(
       GoogleCloudStorageOptions options,
       ExecutorService backgroundTasksThreadPool,
-      Credential credential) {
+      Credentials credentials,
+      boolean useDirectpath) {
```

Comment:
Is an intention to use this constructor directly outside of GCS connector?

Reply:
No this is not about making `StorageStubProvider` public but this is required to incorporate this with Beam. (FYI, I updated the description of PR to have a few pointers to beam code) It passes `HttpRequestInitializer` when creating `GoogleCloudStorageImpl`. Thing is it uses `Credentials` for `HttpRequestInitializer`. At worse, it actually wrapped it with its own private class.

Reply:
So next thing we need to figure out is how beam let gcsio know if it uses a service default account. This can be done by either passing its `Credentials` object or explicit intention to use directpath. Both are possible but I like the former because it can be changed later when directpath supports other credentials in future.

Reply:
Therefore Credentials needs to be added to the public api of `GoogleCloudStorageImpl`. I guess `GoogleCloudStorageOptions` eventually needs it.